internal/task/task/bs/add_spdk_target.go

```diff
@@ -0,0 +1,131 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: CurveAdm
+ * Created Date: 2022-02-08
+ * Author: Jingli Chen (Wine93)
+ */
+
+package bs
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/opencurve/curveadm/cli/cli"
+	comm "github.com/opencurve/curveadm/internal/common"
+	"github.com/opencurve/curveadm/internal/configure"
+	"github.com/opencurve/curveadm/internal/task/scripts"
+	"github.com/opencurve/curveadm/internal/task/step"
+	"github.com/opencurve/curveadm/internal/task/task"
+)
+
+type SpdkTargetOption struct {
+	Host           string
+	User           string
+	BinaryPath     string
+	Cpumask        string
+	Sockname       string
+	ISCSIlogpath   string
+	Create         bool
+	Tid            string
+	Volume         string
+	OriginBdevname string
+	CurveBdevname  string
+	RpcPath        string
+	Size           uint64
+	Blocksize      uint64
+	CbdPath        string
+	IsExcusive     bool
+	TargetIP       string
+	TargetPort     uint64
+	InitiatorIP    string
+	InitiatorMask  uint64
+	VolumeName     string
+}
+
+func NewAddSpdkTargetTask(curveadm *cli.CurveAdm, cc *configure.ClientConfig) (*task.Task, error) {
+	options := curveadm.MemStorage().Get(comm.KEY_SPDK_TARGET_OPTIONS).(SpdkTargetOption)
+	user, volume := options.User, options.Volume
+	hc, err := curveadm.GetHost(options.Host)
+	if err != nil {
+		return nil, err
+	}
+
+	subname := fmt.Sprintf("host=%s volume=%s", options.Host, volume)
+	t := task.NewTask("Add SpdkTarget", subname, hc.GetSSHConfig())
+
+	// add step
+	var output string
+	containerId := DEFAULT_SPDK_TGT_CONTAINER_NAME
+	targetScriptPath := "/curvebs/tools/sbin/add_spdk_target.sh"
+	targetScript := scripts.TARGET
+	cmd := fmt.Sprintf("/bin/bash %s %s %s %s %s %s %d %s %v %s %d %s %d %s %s %v %d",
+		targetScriptPath, options.Sockname, options.Volume,
+		options.OriginBdevname, options.CurveBdevname,
+		options.RpcPath, options.Blocksize, options.CbdPath,
+		options.IsExcusive, options.TargetIP, options.TargetPort,
+		options.InitiatorIP, options.InitiatorMask,
+		user, options.Create, options.Size, options.VolumeName)
+	toolsConf := fmt.Sprintf(FORMAT_TOOLS_CONF, cc.GetClusterMDSAddr())
```

Comment:
ditto. If there are some parameters that are not required, then these need to be reconsidered